### ISSUE TITLE
onChange typescript definition problem

### ISSUE THREAD

**Issue by zwigglers** [buildo/react-autosize-textarea#137]
There is a problem with the typescript definition for onChange.
```
<textarea onChange={(event) => setState({value: event.target.value})} />
```
textarea onChange event: **React.ChangeEvent\<HTMLTextAreaElement>**

```
<TextareaAutosize onChange={(event) => setState({value: event.target.value})} />
```
TextareaAutosize onChange event: **React.FormEvent\<HTMLTextAreaElement>**

Consequently, TextareaAutosize event definition is missing out event.target.value

**Comment by veej** [comment]
Hi @zwigglers, thanks for reporting this issue.
When dealing with events it's always preferable to use `currentTarget` instead of `target`
(you can find more details [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239) and [here](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11508#issuecomment-256045682)).

TL;DR: the reason why `event.target.value` is missing out from the react typings is that, in general, it's impossible to statically type `event.target` (since, in some cases, it could return a child of the component to which the handler is actually attached). Using `currentTarget` (which is guaranteed to always return the component that is handling the event), is generally a safer and better-typed option.

I'll keep this issue open just to replace `e.target` with `e.currentTarget` in the examples.
